include/triton-shared/Conversion/TritonArithToLinalg/ConversionPatterns.hpp

```diff
@@ -836,8 +836,7 @@ struct AssertConverter : public OpConversionPattern<triton::AssertOp> {
     }
 
     auto assertMessage =
-        llvm::formatv("{0}.py:{1}: {2} Assertion `{3}` failed", op.getFile(),
-                      op.getLine(), op.getFunc(), op.getMessage());
+        llvm::formatv("{0}.py:{1}: {2} Assertion `{3}` failed", op.getMessage());
```

Comment:
ah good catch. @zhaoshiz were there any issues with the other arguments that required you to remove them?

Reply:
This is the error shown in the first comment.  Triton AssertOp's no longer have these functions, only a message.

cf. https://github.com/triton-lang/triton/commit/185ad5960635dc9ccdd1ed8a20240e7a4b11fd12

Reply:
yea, good catch, I should look more carefully. Yes, Triton has removed File, Line, Func arguments from it's AssertOp

Reply:
looks like we have a few failures. would you mind taking a look at those? you can xfail the ones in Conversion/TritonToLinalg because we're retiring the pass soon. thank you!

Reply:
> [...] Conversion/TritonToLinalg because we're retiring the pass soon. thank you!

What's the plan?  Replace it with Conversion/TritonToLinalgExperimental?

Reply:
that's correct. we will be using Conversion/TritonToLinalgExperimental going forward. we will update the name to remove the "experimental" suffix once the original TritonToLinalg pass is removed.

Reply:
> looks like we have a few failures. would you mind taking a look at those? you can xfail the ones in Conversion/TritonToLinalg because we're retiring the pass soon. thank you!

yes, I'm looking through those test failures.

Reply:
I managed to fix/xfail most of the failures except:
  TRITON-SHARED :: Conversion/StructuredToMemref/wraparound_side_by_side.mlir
  TRITON-SHARED :: Conversion/StructuredToMemref/wraparound_stacked.mlir

The assertion is caused by LLVM commit https://github.com/llvm/llvm-project/commit/889b67c9d30e3024a1317431d66c22599f6c2011, which consideres <2x?> and <?x?> are mismatch while both are dynamic shapes. I asked if that's intended:
https://github.com/llvm/llvm-project/commit/889b67c9d30e3024a1317431d66c22599f6c2011#r149804171

Reply:
I fixed the failure StructuredToMemref/get_num_programs.mlir: https://github.com/microsoft/triton-shared/pull/187/commits/2153c535be210e50f9e5196e5b04f2bac9c03c6f after syncing-up with  Triton@acc25d91fba850c18c099e7e577962ba56bdd06c and
LLVM@86b69c31642e98f8357df62c09d118ad1da4e16a.

Reply:
thank you! looks like we have some failure in the "Prepare CPU backend" step. Doesn't look immediately clear to me what could be the cause. Perhaps the path that triton uses to store llvm has changed in the newer commits?

Reply:
hmm... I did see a symlink to my llvm being created in `${HOME}/.triton/llvm/`, and the two commands below worked fine in my local env:
  `file $(pwd)/build/$(ls $(pwd)/build | grep -i cmake)/third_party/triton_shared/tools/triton-shared-opt/triton-shared-opt`
  `ls ${HOME}/.triton/llvm/$(ls ${HOME}/.triton/llvm/ | grep -i llvm)/bin`

is there a verbose log of gh workflow?